Report/sections/2_design/code_snippet.tex

```diff
@@ -41,4 +53,12 @@ \section{Code examples}
 }
 \end{lstlisting}
 
-Code snippet~\ref{lst:SampleScreen1} gives an example of a screen. The contents thereof are explained below.
\ No newline at end of file
+Code \snipref{lst:SampleScreen1} shows an example of a screen. A screen represents a single area within the game, and each of the functions used within the \texttt{Screen} block define the layout, what happens when the screen is loaded and the entities that should be present as well as the exits of the screen. 
+
+When a ScreenExit is created, one must first define the direction of the exit, for example the bottom of the screen as shown on line 48.
```

Comment:
Use the escape character to create a label inside the code and refer to it here. Then we don't have to update "line ..." every time we update the snippet :)